lib/utils/parsing_and_formatting.py

```diff
@@ -176,3 +186,11 @@ def parse_logical_operator(logic_op, idx, num_filters):
         raise ValueError(f"Input logical operator in filter condition {idx} must be one of: "
                          ", ".join(["\'" + str(term) + "\'" for term in AQL_logical_operators]))
     return logic_op.upper()
+
+
+def partition_controlled_parsed_filters(parsed_filters):
+    # separates out controlled parsed_filters from uncontrolled for validation
+    uc_filters = [pf for pf in parsed_filters if pf['field'].startswith('custom:')]
+    c_filters = [cf for cf in parsed_filters if cf not in uc_filters]
+
+    return c_filters, uc_filters
```

Comment:
`c_filters` `uc_filters` are used throughout and are slightly opaque as to meaning without context. The curse of variable size. Perhaps `controlled_filters` and `custom_filters` if that doesn't hurt readability too much due to length.